### DISCUSSION TITLE
cd on quit in nnn wrapper script?

### DISCUSSION THREAD

**Discussion by rieje** [jarun/nnn#1863] [Q&A] [answered]
Is there a way to get cd on quit functionality (Ctrl-g only) when using nnn in a wrapper script instead of a shell function? I copied the contents of the function into the script  and it doesn't seem to work.

**Comment by N-R-K** [comment, marked as answer]
> using nnn in a wrapper script instead of a shell function?

Not possible. A process cannot arbitrarily change the working directory of another process.

**Reply by rieje** [reply]
I suppose an elegant way for nnn to support this is an option to output the working directory on exit, i.e. in a script `cd "$(nnn)"`.

**Reply by 0xACE** [reply]
`.lastd` contains the last directory, but you still need to instruct your shell to read it, just use the function...
